mod/discussions/classes/Elgg/Discussions/Upgrades/MigrateDiscussionReply.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace Elgg\Discussions\Upgrades;
+
+use \Elgg\Upgrade\Batch;
+use \Elgg\Upgrade\Result;
+use \Elgg\Database\Update;
+
+/**
+ * Migrate 'object', 'discussion_reply' to 'object', 'comment'
+ *
+ * @since 3.0
+ */
+class MigrateDiscussionReply implements Batch {
+	
+	/**
+	 * {@inheritDoc}
+	 */
+	public function getVersion() {
+		return 2017112800;
+	}
+	
+	/**
+	 * {@inheritDoc}
+	 */
+	public function shouldBeSkipped() {
+		return empty($this->countItems());
+	}
+	
+	/**
+	 * {@inheritDoc}
+	 */
+	public function countItems() {
+		
+		$hidden = access_show_hidden_entities(true);
+		
+		$count = elgg_get_entities([
+			'type' => 'object',
+			'subtype' => 'discussion_reply',
+			'count' => true,
+		]);
+		
+		access_show_hidden_entities($hidden);
+		
+		return $count;
+	}
+	
+	/**
+	 * {@inheritDoc}
+	 */
+	public function needsIncrementOffset() {
+		return false;
+	}
+	
+	/**
+	 * {@inheritDoc}
+	 */
+	public function run(Result $result, $offset) {
+		
+		$qb = Update::table('entities', 'e')
+			->set('e.subtype', '"comment"')
```

Comment:
Just as a note, one of the reasons we use QB is not to have to do the escaping:

```php
$qb = Update::table('entities', 'e);
$qb->set('e.subtype', $qb->param('comment', 'string'))
->where($qb->compare('e.subtype', '=', 'discussion_reply', 'string'));
```

Reply:
thanks will try to remember for next time